hack/verify-dirty.sh

```diff
@@ -8,36 +8,39 @@ echo "#############################"
 echo "Verify uncommitted files..."
 echo "#############################"
 
-# Option to ignore changes to compatibility image path in
-# tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go and plugin runtime version in
-# cli/runtime/version/zz_generated_plugin_runtime_version.go. This is to support `usebom` directive
-# in tests.
-ignore_files=(':(exclude)tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go' ':(exclude)cli/runtime/version/zz_generated_plugin_runtime_version.go')
-# Temporarily excluding UI generated bindata file from verify (zz_generated.bindata.go). Currently running into issues
-# blocking the CI main build. CI generated bindata is different from bindata file generated on local
-# developer machines, causing this failure. Need to root cause and then remove this exclusion.
-ignore_file_ui_bindata=':!tkg/manifest/server/zz_generated.bindata.go'
-
-if ! (git diff --quiet HEAD -- . "${ignore_files[@]}" "${ignore_file_ui_bindata}"); then
+if ! (git diff --quiet HEAD -- .); then
    echo -e "\nThe following files are uncommitted. Please commit them or add them to .gitignore:";
-   git diff --name-only HEAD -- . "${ignore_files[@]}" "${ignore_file_ui_bindata}" | awk '{print "- " $0}'
+   git diff --name-only HEAD -- . | awk '{print "- " $0}'
    echo -e "\nDiff:"
-   git --no-pager diff  HEAD -- . "${ignore_files[@]}" "${ignore_file_ui_bindata}"
+   git --no-pager diff  HEAD -- .
    exit 1
 else
    echo "OK"
 fi
 
-
 echo
 echo "#############################"
 echo "Verify make package-vendir-sync..."
 echo "#############################"
 make package-vendir-sync
-if ! (git diff --quiet HEAD -- . "${ignore_files[@]}" "${ignore_file_ui_bindata}"); then
+echo "Done vendir sync!"
+
+# This is required for github actions execution
+# because the package-vendir-sync operation renders
+# the pakcage directory contents inaccessible to the
+# host machine user.
+if [[ -n "${GITHUB_ACTIONS}" ]]; then
+  echo "Resetting directory ownership"
+  sudo chown -R $(id -nu):$(id -ng) .;
```

Comment:
This is a great find! Is it possible to do this step in the workflow yaml instead of here? I found https://github.com/marketplace/actions/reset-workspace-ownership-action, but we could do this ourselves in a `step`?

I also found this: https://github.com/marketplace/actions/reset-workspace-ownership-action

Reply:
It'd require an amount of refactoring effort in the verify target flow, since this step needs to be done exactly after `package-vendir-sync` and before running `git diff`. Having it this way is simpler and also provides consistent behaviour if run on developer machines.

Reply:
I'm okay with the current approach..